tests/test_globio.py

```diff
@@ -451,10 +451,7 @@ def test_missing_field_in_msa_parameters(self):
             file.write('1,2\n')
         validation_warnings = globio.validate(
             {'msa_parameters_path': msa_parameters_path})
-        expected_message = ("Fields are missing from this table: "
-                            "['MEASUREMENT', 'MSA_TYPE', 'MSA_X', 'SE', "
-                            "'VALUE']")
-        actual_messages = set()
-        for keys, error_strings in validation_warnings:
-            actual_messages.add(error_strings)
-        self.assertTrue(expected_message in actual_messages)
+        expected_warning = (
+            ['msa_parameters_path'],
+            'msa_type matched 0 headers, expected at least one')

```

Comment:
That's a good point. I'm not totally sure how to accomplish this since, in the cases that use a more complicated regex, that _is_ the identifier - there's no alternate user-friendly name for the column. But the large majority are an ordinary string.

Reply:
This is hopefully resolved: the new message is `'Expected the %s "%s" but did not find it' % (header type, header name)` e.g. 'Expected the column "lucode" but did not find it'. Including the header type (row, column, or field) should make the message clearer. 
And as we discussed, I've done away with the regexes, and validation.py will not try to validate user-defined headers at all. So the header name in the message will always say exactly what's missing.